include/natalie/io_object.hpp

```diff
@@ -20,11 +20,18 @@ namespace ioutil {
     StringObject *convert_using_to_path(Env *env, Value path);
     int object_stat(Env *env, Value io, struct stat *sb);
     struct flags_struct {
+        enum class read_mode { none,
+            text,
+            binary };
```

Comment:
I did not write it with this indentation, clang-format did. We might want to tweak the settings a bit for this one, but I don't think that should be blocking for this feature.